sycl/doc/extensions/proposed/sycl_ext_intel_dataflow_pipes.asciidoc

```diff
@@ -254,6 +270,26 @@ Type aliases in {cpp} through the `using` mechanism do not change the type of a
   pipe<type_alias, int>::write(0);
 ----
 
+== Host pipe read/write
+
+The read/write member functions of a host pipe have different signatures when they are called from the host side, in which case a `sycl::queue` is added to the parameters.
+
+[source,c++,Host pipe read write members,linenums]
+----
+template <typename name,
+          typename dataT,
+          typename propertiesT = properties<>>
+class pipe {
+  // Blocking
+  static dataT read( const queue &q, memory_order order = memory_order::seq_cst );
+  static void write( const queue &q, const dataT &data, memory_order order = memory_order::seq_cst );
+
+  // Non-blocking
+  static dataT read( const queue &q, bool &success_code, memory_order order = memory_order::seq_cst );
+  static void write( const queue &q, const dataT &data, bool &success_code, memory_order order = memory_order::seq_cst );
```

Comment:
The queue parameter probably shouldn't be const, because the function will need to call q.submit for enqueueing an operation that calls the opencl function.

Reply:
@pcolberg You can close this one